### PR TITLE
statusbar plugin: adds run energy and inventory weight

### PR DESCRIPTION
**Pull request by skibz**
hello!

i decided to share my changes to the statusbar plugin with you all.
it adds a new bar that works like the hitpoints and prayer ones, and also shows your inventory weight.

the run energy bar will only render when the client is in resizable mode, as i couldn't find a neat way to include it in fixed mode. moreover, in fixed mode, the regular run energy indicator is a lot closer to the other two bars, arguably obviating the need for it entirely.

a small summary of what has been changed in the plugin:

* adding the run energy bar
* adding support for the healing overlay to run energy bar
* adding icons for run and weight

i'd love to get some feedback and indication of whether you'd be keen to have this included!

eg

thanks y'all <3

### CODE REVIEW

**Comment by raiyni** [comment]
You need to format this with our formatter settings. https://github.com/runelite/runelite/wiki/Code-Conventions

**Comment by skibz** [comment]
thanks @raiyni , reasonably certain i've got it right now.

**Comment by skibz** [comment]
oh, nevermind :P

**Comment by skibz** [comment]
i forgot to add some transparency to the bar colour... sorted!

transparency

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java [hunk 1]
```diff
@@ -215,83 +334,65 @@ else if (poisonState > 0)
...
+			final String counterWeightText = Integer.toString(counterWeight);
 
 			renderIconsAndCounters(g, offsetPrayerX, offsetPrayerY, prayerImage, counterPrayerText, PRAYER_LOCATION_X);
 			renderIconsAndCounters(g, offsetHealthX, offsetHealthY, healthImage, counterHealthText, HEALTH_LOCATION_X);
-		}
-
-		return null;
-	}
-
-	private static void renderBar(Graphics2D graphics, int x, int y, int max, int current, int height, Color filled)
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java:241]: Don't rearrange these methods

Comment (skibz) [reply runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java]: roger

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java [hunk 2]
```diff
@@ -32,29 +32,29 @@
...
-		name = "Show hitpoints & prayer counter",
-		description = "Shows current amount of hitpoints & prayer on the status bars"
+			keyName = "enableCounter",
+			name = "Show status counters",
+			description = "Shows current amount of hitpoints, prayer, energy and weight on the status bars"
 	)
 	default boolean enableCounter()
 	{
-		return false;
+		return true;
```
Comment (dekvall) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java:41]: Why did you change this default?

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java [hunk 3]
```diff
@@ -32,29 +32,29 @@
 public interface StatusBarsConfig extends Config
 {
 	@ConfigItem(
-		keyName = "enableCounter",
-		name = "Show hitpoints & prayer counter",
-		description = "Shows current amount of hitpoints & prayer on the status bars"
+			keyName = "enableCounter",
+			name = "Show status counters",
+			description = "Shows current amount of hitpoints, prayer, energy and weight on the status bars"
```
Comment (dekvall) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java:37]: Set your continuation indent to 4

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java [hunk 4]
```diff
@@ -49,6 +45,11 @@
 import net.runelite.client.ui.overlay.components.TextComponent;
 import net.runelite.client.util.ImageUtil;
 
+import javax.inject.Inject;
+import java.awt.*;
```
Comment (dekvall) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java]: No wildcard imports please

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java [hunk 5]
```diff
@@ -84,9 +88,16 @@
...
+	private final BufferedImage weightImage;
 
 	@Inject
-	private StatusBarsOverlay(Client client, StatusBarsConfig config, SkillIconManager skillIconManager, ItemStatChangesService itemstatservice)
+	private StatusBarsOverlay(
+			Client client,
+			StatusBarsConfig config,
+			SkillIconManager skillIconManager,
+			ItemStatChangesService itemstatservice
+	)
```
Comment (dekvall) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java:102]: It looks like this was just formatted, can you change it back?

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java [hunk 6]
```diff
@@ -171,11 +196,41 @@ else if (poisonState > 0)
 		final int quickPrayerState = client.getVar(Varbits.QUICK_PRAYER);
 		final Color prayerBar = quickPrayerState == 1 ? QUICK_PRAYER_COLOR : PRAYER_COLOR;
 
-		renderBar(g, offsetHealthX, offsetHealthY,
-			maxHealth, currentHealth, height, healthBar);
-
-		renderBar(g, offsetPrayerX, offsetPrayerY,
-			maxPrayer, currentPrayer, height, prayerBar);
+		final int maxEnergy = 100;
```
Comment (dekvall) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java:201]: This should be static

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java [hunk 7]
```diff
@@ -49,6 +44,12 @@
 import net.runelite.client.ui.overlay.components.TextComponent;
 import net.runelite.client.util.ImageUtil;
 
+import javax.inject.Inject;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
```
Comment (dekvall) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java:51]: Can you move these back where they were?

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java [hunk 8]
```diff
@@ -171,11 +198,41 @@ else if (poisonState > 0)
...
+				g,
+				offsetPrayerX,
+				offsetPrayerY,
+				maxPrayer,
+				currentPrayer,
+				height,
+				prayerBar
+		);
+
+		if (resizedBottomMode)
```
Comment (dekvall) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java:224]: Maybe put a comment on why we are only rendering this in `resizedBottomMode`

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java [hunk 9]
```diff
@@ -129,23 +144,35 @@ public Dimension render(Graphics2D g)
...
 			height = HEIGHT;
+
 			offsetHealthX = (location.getX() - offsetLeft.getX());
 			offsetHealthY = (location.getY() - offsetLeft.getY());
+
 			offsetPrayerX = (location.getX() - offsetRight.getX()) + curWidget.getWidth();
 			offsetPrayerY = (location.getY() - offsetRight.getY());
+
+			offsetEnergyX = 0;
+			offsetEnergyY = 0;
```
Comment (dekvall) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java:175]: These are never used right?

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java [hunk 10]
```diff
@@ -215,24 +278,62 @@ else if (poisonState > 0)
...
+			}
 		}
 
 		if (config.enableSkillIcon() || config.enableCounter())
 		{
 			final BufferedImage healthImage = skillIconManager.getSkillImage(Skill.HITPOINTS, true);
 			final int counterHealth = client.getBoostedSkillLevel(Skill.HITPOINTS);
 			final int counterPrayer = client.getBoostedSkillLevel(Skill.PRAYER);
+			final int counterEnergy = client.getEnergy();
+			final int counterWeight = client.getWeight();
```
Comment (dekvall) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java:306]: Not a big fan of this naming, since counterweight is something else, but i guess it follows suite.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java [hunk 11]
```diff
@@ -129,23 +144,35 @@ public Dimension render(Graphics2D g)
 		final Point offsetLeft = curViewport.getOffsetLeft();
 		final Point offsetRight = curViewport.getOffsetRight();
 		final Point location = curWidget.getCanvasLocation();
-		final int height, offsetHealthX, offsetHealthY, offsetPrayerX, offsetPrayerY;
+		final boolean resizedBottomMode = curViewport == Viewport.RESIZED_BOTTOM;
+
+		final int height, offsetHealthX, offsetHealthY, offsetPrayerX, offsetPrayerY, offsetEnergyX, offsetEnergyY;
```
Comment (dekvall) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java:149]: Don't move this

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java (hunk 2 above)
Comment (skibz) [reply runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java]: i just forgot to change this back...

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java (hunk 9 above)
Comment (skibz) [reply runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java]: they have to be initialised, unfortunately. i've moved it outside of the conditionals.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java (hunk 10 above)
Comment (skibz) [reply runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java:352]: well, feel free to suggest something...

counterInventoryWeight?
counterMass?

**Comment by skibz** [comment]
i ought to have checked the bug tracker before opening this. i didn't realise somebody else was also proposing changes to this plugin.

sorry about that :)

**Comment by makingcomment** [comment]
I think #7730 is a better implementation of a similar idea.
